library/ssl_tls.c

```diff
@@ -245,6 +245,25 @@ int mbedtls_ssl_session_copy( mbedtls_ssl_session *dst,
     return( 0 );
 }
 
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+static int resize_buffer( unsigned char **buffer, size_t len, size_t *len_info )
+{
+    unsigned char* resized_buffer = mbedtls_calloc( 1, len );
+    if( resized_buffer == NULL )
+        return -1;
+
+    /* Copy the smaller one out of old and new buffer lengths. */
+    memcpy( resized_buffer, *buffer, ( len < *len_info ) ? len : *len_info );
```

Comment:
I'm not sure when we would use `*len_info` as the length of the copy, or if the previous `mbedtls_calloc()` might be too large if it is always `len`. Could you expand a bit more in the C comment, please?

Reply:
When downsizing the buffer, len is smaller, and when upsizing - len_info. I'll expand the comment.

Reply:
Done.

Reply:
Yes, I should consider this function used in both ways. OK. Much clearer now in the code. Thanks.